src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -0,0 +1,2271 @@
+				} catch (OHServiceException e) {
+					OHServiceExceptionUtil.showMessages(e);
+					MessageDialog.info(null, "angal.inventory.pleasevalidateinventoryagainsbeforeconfirmation.msg");
+					confirmButton.setEnabled(false);
+					return;
+				}
+			}
+		});
+		return confirmButton;
+	}
+
+	private String checkParamsValues(String chargeCode, String dischargeCode, Integer supplierId, String wardCode) {
+		if (chargeCode == null || chargeCode.isEmpty()) {
+			return "angal.inventory.choosechargetypebeforevalidation.msg";
+		}
+		if (dischargeCode == null || dischargeCode.isEmpty()) {
+			return "angal.inventory.choosedischargetypebeforevalidation.msg";
+		}
+		if (supplierId == null || supplierId == 0) {
+			return "angal.inventory.choosesupplierbeforevalidation.msg";
+		}
+		if (wardCode == null || wardCode.isEmpty()) {
+			return "angal.inventory.choosedestinationbeforevalidation.msg";
+		}
+		return null;
+	}
+
+	private JScrollPane getScrollPaneInventory() {
+		if (scrollPaneInventory == null) {
+			scrollPaneInventory = new JScrollPane();
+			try {
+				scrollPaneInventory.setViewportView(getJTableInventoryRow());
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+			}
+		}
+		return scrollPaneInventory;
+	}
+
+	private JTable getJTableInventoryRow() throws OHServiceException {
+		if (jTableInventoryRow == null) {
+			jTableInventoryRow = new JTable();
+			jTextFieldEditor = new JTextField();
+			jTableInventoryRow.setFillsViewportHeight(true);
+			model = new InventoryRowModel();
+			jTableInventoryRow.setModel(model);
+			jTableInventoryRow.setAutoCreateColumnsFromModel(false);
+			for (int i = 0; i < columnVisible.length; i++) {
+				jTableInventoryRow.getColumnModel().getColumn(i).setPreferredWidth(columwidth[i]);
+				if (!columnVisible[i]) {
+					jTableInventoryRow.getColumnModel().getColumn(i).setMinWidth(0);
+					jTableInventoryRow.getColumnModel().getColumn(i).setMaxWidth(0);
+					jTableInventoryRow.getColumnModel().getColumn(i).setPreferredWidth(0);
+				}
+				if (columnCentered[i]) {
+					jTableInventoryRow.getColumnModel().getColumn(i).setCellRenderer(new CenterTableCellRenderer());
+				}
+				if (columnDecimalNumber[i]) {
+					jTableInventoryRow.getColumnModel().getColumn(i).setCellRenderer(new DecimalNumberTableCellRenderer());
+				}
+			}
+			jTableInventoryRow.getSelectionModel().addListSelectionListener(new ListSelectionListener() {
+
+				@Override
+				public void valueChanged(ListSelectionEvent e) {
+					if (e.getValueIsAdjusting()) {
+						int selectedRow = jTableInventoryRow.getSelectedRow();
+						if (selectedRow != -1) {
+							MedicalInventoryRow medInvRow = (MedicalInventoryRow) jTableInventoryRow.getValueAt(selectedRow, -1);
+							Lot lot = medInvRow.getLot();
+							if (lot == null) {
+								lotButton.setEnabled(true);
+							} else {
+								BigDecimal cost = lot.getCost() != null ? lot.getCost() : BigDecimal.ZERO;
+								if (lot != null) {
```

Comment:
This statement always is true.   If not the previous  line would throw a NPE.  Besides line 1195 already tests for `null` and if so takes the `then` path.